xonsh/completer.py

```diff
@@ -78,16 +78,54 @@ def _normpath(p):
 
     return p
 
-class Completer(object):
-    """This provides a list of optional completions for the xonsh shell."""
+class KnownCommands(object):
```

Comment:
This could be `KnownCommandsCache` or `CommandsCache` or similar, but I don't think it matters much.